crates/store/re_types_core/src/as_components.rs

```diff
@@ -34,7 +33,37 @@ pub trait AsComponents {
     //
     // NOTE: Don't bother returning a CoW here: we need to dynamically discard optional components
     // depending on their presence (or lack thereof) at runtime anyway.
-    fn as_component_batches(&self) -> Vec<ComponentBatchCowWithDescriptor<'_>>;
+    #[deprecated(since = "0.22.0", note = "use as_serialized_batches instead")]
```

Comment:
It's worth to also change the docstring to `DEPRECATED - DO NOT USE` or similar.